sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuiteBase.scala

```diff
@@ -126,4 +127,512 @@ trait ShowTablesSuiteBase extends QueryTest with DDLCommandTestUtils {
+             |Table: $table
+             |Owner: ${Utils.getCurrentUserName()}
+             |Type: MANAGED
+             |Provider: hive
+             |Serde Library: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
+             |InputFormat: org.apache.hadoop.mapred.TextInputFormat
+             |OutputFormat: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
+             |Storage Properties: [serialization.format=1]
+             |Partition Provider: Catalog
+             |Partition Columns: [`id`]
+             |Schema: root
+             | |-- data: string (nullable = true)
+             | |-- id: long (nullable = true)""".stripMargin
+        assert(actualResult_0_3 === expectedResult_0_3_v1 ||
+          actualResult_0_3 === expectedResult_0_3_v2 ||
+          actualResult_0_3 === expectedResult_0_3_hive)
+
+        assert(result.collect()(1).length == 4)
+        assert(result.collect()(1)(1) === table1)
+        assert(result.collect()(1)(2) === false)
+        val actualResult_1_3 = exclude(result.collect()(1)(3).toString)
+
+        // exclude "Created Time", "Last Access", "Created By", "Location"
+        val expectedResult_1_3_v1 =
+          s"""Catalog: $catalog
+             |Database: $namespace
+             |Table: $table1
+             |Type: MANAGED
+             |Provider: parquet
+             |Partition Provider: Catalog
+             |Partition Columns: [`id1`]
+             |Schema: root
+             | |-- data1: string (nullable = true)
+             | |-- id1: long (nullable = true)""".stripMargin
+        val expectedResult_1_3_v2 =
+          s"""Namespace: $namespace
+             |Table: $table1
+             |Type: MANAGED
+             |Provider: _
+             |Owner: ${Utils.getCurrentUserName()}
+             |Partition Provider: Catalog
+             |Partition Columns: [`id1`]
+             |Schema: root
+             | |-- data1: string (nullable = true)
+             | |-- id1: long (nullable = true)""".stripMargin
+
+        // exclude "Table Properties"
+        val expectedResult_1_3_hive =
+          s"""Catalog: $catalog
+             |Database: $namespace
+             |Table: $table1
+             |Owner: ${Utils.getCurrentUserName()}
+             |Type: MANAGED
+             |Provider: hive
+             |Serde Library: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
+             |InputFormat: org.apache.hadoop.mapred.TextInputFormat
+             |OutputFormat: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
+             |Storage Properties: [serialization.format=1]
+             |Partition Provider: Catalog
+             |Partition Columns: [`id1`]
+             |Schema: root
+             | |-- data1: string (nullable = true)
+             | |-- id1: long (nullable = true)""".stripMargin
+        assert(actualResult_1_3 === expectedResult_1_3_v1 ||
+          actualResult_1_3 === expectedResult_1_3_v2 ||
+          actualResult_1_3 === expectedResult_1_3_hive)
```

Comment:
Don't think this good check. Let's follow existing convention, and create separate tests in V2, V1 and Hive traits.

Reply:
Okay, Let's separate it.